tests/robustness/watch.go

```diff
@@ -70,7 +71,7 @@ func watchUntilRevision(ctx context.Context, t *testing.T, c *traffic.RecordingC
 	var lastRevision int64
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
-	watch := c.Watch(ctx, "", 1, true, true)
+	watch := c.Watch(ctx, "", 1, true, true, cfg.withPrevKV)
```

Comment:
No, need to pass `withPrevKV` here, it's enough that Kubernetes Traffic uses PrevKV. We can pass `false` here.

Reply:
@serathius not sure I fully understand here - the test runner calls into a common watch method for both Kubernetes and non kubernetes traffic, top of the call stach being here: 
https://github.com/etcd-io/etcd/blob/43d675997772bee9cf9d6bd4804ef05cc0cecb01/tests/robustness/main_test.go#L134

Reply:
Please address the comment

Reply:
Even if you enable WithPrevKey on all watches, as we established in previous comments, etcd doesn't guarantee that it will provide it. So the issue with merging watch history could still happen.

The proper solution would be to not use `PrevValue` when merging watch history. It's redundant and not needed. I would recommend that `mergeWatchEventHistory` should not care about PreviousKV and returns `[]PersistedEvent` and not `[]WatchEvent`.

Reply:
Not sure I follow, output of `mergeWatchEventHistory` serves as authoritative list of events that happened. We don't need prevKV there. Not sure what you mean by `otherwise me would have to patch that before passing it into the validate funcs`. The prevKV in eventHistory is not needed for for validators. It just a list of operations in order we believe that were persisted by etcd. 

Changing it might require a bigger refactor, so we can skip it for this PR.

Reply:
Yeah that's what I meant - in order to refactor it, we would have to change what the model operates on as well. I can send a follow up after this one